### PR TITLE
[SYCL][DOC] Move SPIR-V and OpenCL extensions

### PR DESCRIPTION
**Pull request by gmlueck**
The SPIR-V and OpenCL extensions are not SYCL extensions, rather they
are extensions to underlying components used by DPC++.  Therefore, we
consider them part of the design, so move them to "doc/design".

Our policy is to remove SPIR-V and OpenCL extension specifications from
this repo once they are added to the Khronos registry.  The
"SPV_INTEL_arbitrary_precision_int" extension was already added there,
so remove it from this repo.

### CODE REVIEW

File: sycl/doc/extensions/README.md [hunk 1]
```diff
@@ -7,8 +7,6 @@ DPC++ extensions status:
 
 |  Extension  |    Status   |   Comment   |
 |-------------|:------------|:------------|
-| [SPV_INTEL_function_pointers](SPIRV/SPV_INTEL_function_pointers.asciidoc)                                                   | Supported(OpenCL: CPU, GPU; HOST)         | |
-| [SPV_INTEL_inline_assembly](SPIRV/SPV_INTEL_inline_assembly.asciidoc)                                                       | Supported(OpenCL: GPU)                    | |
```
Comment (bader) [inline sycl/doc/extensions/README.md:10]: It it the right time to remove this file?

**Review by bader — APPROVED** [review]

File: sycl/doc/extensions/README.md (hunk 1 above)
Comment (gmlueck) [reply sycl/doc/extensions/README.md:10]: I am preparing another commit with replaces this file, and also adds some more README files.  If you want this file removed sooner, I can remove it in this PR.  Do you want that, or is it OK to wait for my next PR?

**Review by bashbaug — APPROVED** [review]
LGTM!

File: sycl/doc/extensions/README.md (hunk 1 above)
Comment (bader) [reply sycl/doc/extensions/README.md:10]: Next PR sounds good to me.

**Comment by bader** [comment]
@gmlueck, please, fix "Generate Doxygen documentation" job.

**Comment by gmlueck** [comment]
> @gmlueck, please, fix "Generate Doxygen documentation" job.

Done in 94c02111767a1ad04772bcca8a6452176461e6e4

**Review by jbrodman — APPROVED** [review]

**Pull request merged**